internal/adapters/api/handler/permission.go

```diff
@@ -49,12 +51,12 @@ func (h *permissionHandler) GetPermissionByID(c *gin.Context) {
 
 	permission, err := h.permissionService.GetPermissionByID(permissionID)
 	if err != nil {
-		h.logger.Errorf("Get Permission By ID: %s", err.Error())
+		h.logger.Errorf("Get Permission By ID %s Failed: %s", permissionID, err.Error())
```

Comment:
for any endpoint with id or that requires to get any entity by id, you should check for 404 error

Reply:
Done